hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -652,7 +653,7 @@ public static HoodieData<HoodieRecord> convertMetadataToColumnStatsRecords(Hoodi
           String partitionPath = deleteFileInfoPair.getLeft();
           String filePath = deleteFileInfoPair.getRight();
 
-          if (filePath.endsWith(HoodieFileFormat.PARQUET.getFileExtension())) {
+          if (filePath.endsWith(HoodieFileFormat.PARQUET.getFileExtension()) || ExternalFilePathUtil.isExternallyCreatedFile(filePath)) {
```

Comment:
@nsivabalan can we just remove this check?

Reply:
I guess there is some gap here wrt log files. 
for log files, we get stats directly from Append Handle and entries are added to col stats using log file name. 
so, during clean commit metadata, we should be deleting both data files and log files. 

https://issues.apache.org/jira/browse/HUDI-7331

I have created a follow up ticket on this. lets fix it thoroughly.